src/test/java/com/checkmarx/flow/cucumber/integration/cli/CommonCliSteps.java

```diff
@@ -0,0 +1,114 @@
+import java.lang.reflect.InvocationTargetException;
+
+@SpringBootTest(classes = {CxFlowApplication.class, JiraTestUtils.class})
+@Slf4j
+@RequiredArgsConstructor
+public class CommonCliSteps {
+
+    private final IntegrationTestContext testContext;
+    private final FlowProperties flowProperties;
+    private static final String GITHUB_REPO_ARGS = " --repo-url=https://github.com/cxflowtestuser/CLI-Integration-Tests --repo-name=CLI-Integration-Tests --github --branch=master --namespace=cxflowtestuser --blocksysexit";
+    private static final String JIRA_PROJECT = "CIT";
+    private String commandlineConstantArgs;
+
+    @Autowired
+    private JiraProperties jiraProperties;
+
+    @Autowired
+    private IJiraTestUtils jiraUtils;
+
+    @Before("@SAST_CLI_SCAN")
+    public void beforeEachScenario() throws IOException {
+        log.info("Setting bugTracker: Jira");
+        flowProperties.setBugTracker("JIRA");
+        log.info("Jira project key: {}", JIRA_PROJECT);
+        jiraProperties.setProject(JIRA_PROJECT);
+        initJiraBugTracker();
+        testContext.reset();
+    }
+
+    @Given("repository is github")
+    public void initGithubArguments(){
+        commandlineConstantArgs = GITHUB_REPO_ARGS;
+    }
+
+    @Given("repository is github-sca")
+    public void initGithubSCAArguments(){
+        // Need to add the correct SCA repo
+    }
+
+    @After("@SAST_CLI_SCAN")
+    public void afterEachScenario() {
+        log.info("cleaning jira project: {}", jiraProperties.getProject());
+        jiraUtils.cleanProject(jiraProperties.getProject());
+    }
+
+    @When("running CxFlow scan with command line: {string}")
+    public void runningCxFlowWithCommandLineCommandLine(String commandLine) {
+        String fullCommandLine = commandLine + commandlineConstantArgs;
+        log.info("Running CxFlow scan with command line: {}", fullCommandLine);
+        Throwable exception = null;
+        try {
+            TestUtils.runCxFlow(testContext.getCxFlowRunner(), fullCommandLine);
+        } catch (Throwable e) {
+            exception = e;
+        }
+        testContext.setCxFlowExecutionException(exception);
+    }
+
+    @Then("CxFlow exits with exit code {}")
+    public void cxflowExitsWithStatusCodeStatusCode(int expectedExitCode) {
+
+        Throwable exception = testContext.getCxFlowExecutionException();
+
+        Assert.assertNotNull("Expected an exception to be thrown.", exception);
+        Assert.assertEquals(InvocationTargetException.class, exception.getClass());
+
+        Throwable targetException = ((InvocationTargetException) exception).getTargetException();
+        Assert.assertTrue(targetException instanceof ExitThrowable);
+
+        int actualExitCode = ((ExitThrowable) targetException).getExitCode();
+        Assert.assertEquals(expectedExitCode, actualExitCode);
+        log.info("CxFlow exited with expected exit code: {}", expectedExitCode);
+    }
+
+    @And("bugTracker contains {} issues")
```

Comment:
in new code, please use Given/When/Then annotations and not the And And, it will be simpler for programmers to get the context (in the feature file it will stay AND)